server/app/services/rpc_client.rb

```diff
@@ -3,17 +3,17 @@
 class RpcClient
 
   class Error < StandardError
-    attr_accessor :code, :message, :backtrace
+    attr_reader :code
 
-    def initialize(code, message, backtrace = nil)
-      self.code = code
-      self.message = message
-      self.backtrace = backtrace
+    def initialize(code, message, remote_backtrace = nil)
+      @code = code
+      super(message)
+      set_backtrace(caller) unless backtrace
```

Comment:
This feels very complicated, and ruby's Exceptions are confusing enough that I don't know the right way to do this... but could we avoid having to set the exception backtrace ourselves by doing this gluing later?

```ruby
def backtrace
  @remote_backtrace + super
end
```